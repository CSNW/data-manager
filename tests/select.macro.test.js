const pluginTester = require('babel-plugin-tester');
const plugin = require('babel-plugin-macros');

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: [
    `
      import { map } from 'data-manager';
      import select from '../select.macro';

      const name = 'Population'

      map(select('Year', name));
    `,
    `
      import { map } from 'data-manager';
      import select from '../select.macro';

      const name = 'Population'

      map(select(['Year', name]));
    `,
    `
      import { map } from 'data-manager';
      import select from '../select.macro';

      const name = 'Population';
      const state = 'state';
      const tract = 'tract';

      map(select({
        year: 'Year',
        'population': name,
        [state]: state,
        tract
      }));
    `
  ]
});
