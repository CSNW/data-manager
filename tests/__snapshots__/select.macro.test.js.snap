// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 1. macros: 1. macros 1`] = `
"
import { map } from 'data-manager';
import select from '../select.macro';

const name = 'Population'

map(select(['Year', name]));

      ↓ ↓ ↓ ↓ ↓ ↓

import { map } from 'data-manager';
const name = 'Population';
map(row => {
  return {
    'Year': row['Year'],
    [name]: row[name]
  };
});
"
`;

exports[`macros 2. macros: 2. macros 1`] = `
"
import { map } from 'data-manager';
import select from '../select.macro';

const name = 'Population';
const state = 'state';
const tract = 'tract';

map(select({
  year: 'Year',
  'population': name,
  [state]: state,
  tract
}));

      ↓ ↓ ↓ ↓ ↓ ↓

import { map } from 'data-manager';
const name = 'Population';
const state = 'state';
const tract = 'tract';
map(row => {
  return {
    year: row['Year'],
    'population': row[name],
    [state]: row[state],
    tract: row[tract]
  };
});
"
`;
